lesson4/lesson4.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"math"
+)
+
+type Shape interface {
+	Area() float64
+	Perimeter() float64
+}
+type Circle struct {
+	radius float64
+}
+type Rectangle struct {
+	height float64
+	width  float64
+}
+
+func DescribeShape(s Shape) {
+	fmt.Println(s)
+	fmt.Printf("Area: %.2f\n", s.Area())
+	fmt.Printf("Perimeter: %.2f\n", s.Perimeter())
+}
+func (s Circle) String() string {
```

Comment:
if we started to name Circle as "s" and bind methods to "s", then we should use this name with all bidings to this object. This would be good style, I will look through code and will see all the methods of "s".

for example use "s" for all bindings to"Circle" and "a" for all bindings to "Rectangle". It will be more clean code